src/tests/Autosuggest.js

```diff
@@ -412,6 +426,117 @@ describe('Autosuggest', function() {
     });
   });
 
+  describe('onSuggestionFocused', function() {
+    beforeEach(function() {
+      onSuggestionFocused.mockClear();
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbStrings}
+                     onSuggestionFocused={onSuggestionFocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+
+      it('should call onSuggestionFocused when suggestion focused using mouse', function() {
+        mouseOverFromInputToSuggestion(1);
+        expect(onSuggestionFocused).toBeCalledWith('Mordialloc');
+      });
+    });
+
+    describe('Keyboard interactions', function() {
+
+      it('should call onSuggestionFocused when suggestion focused using Up/Down keys', function() {
+        clickDown();
+        expect(onSuggestionFocused).toBeCalledWith('Mill Park');
+      });
+
+      it('should not call onSuggestionFocused when first suggestion is focused and Up key is clicked', function() {
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickUp();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+
+      it('should not call onSuggestionFocused when last suggestion is focused and Down key is clicked', function() {
+        clickDown();
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickDown();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+    });
+  });
+
+  describe('onSuggestionUnfocused', function() {
+    beforeEach(function() {
+      onSuggestionUnfocused.mockClear();
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbObjects}
+                     suggestionRenderer={renderSuburbObject}
+                     suggestionValue={getSuburbObjectValue}
+                     onSuggestionUnfocused={onSuggestionUnfocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+
+      it('should call onSuggestionUnfocused when suggestion unfocused using mouse', function() {
+        mouseOverFromInputToSuggestion(0);
+        mouseOverBetweenSuggestions(0, 1);
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should call onSuggestionUnfocused when suggestion selected with mouse', function() {
+        mouseOverFromInputToSuggestion(0);
+        mouseDownSuggestion(0);
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+    });
+
+    describe('Keyboard interactions', function() {
+
+      it('should call onSuggestionUnfocused when suggestion unfocused using Up/Down keys', function() {
+        clickDown();
+        clickDown();
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
```

Comment:
No. This section is for `onSuggestionUnfocused`. The test above checks that `onSuggestionUnfocused` is called after Up/Down, which is not true in case ESC was pressed before.